### ISSUE TITLE
Strings split on `-and`

### ISSUE THREAD

**Issue by OJFord** [xonsh/xonsh#1921]
Reproduce:
```sh
$ touch fooandbar foo-bar foo-and-bar
```

Expected:
```sh
[No output]
$ ls
foo-and-bar  foo-bar  fooandbar
```

Actual:
```sh
xonsh: subprocess mode: command not found: bar
Did you mean one of the following?
[Snip]
$ ls
foo-  foo-bar  fooandbar
```

Workaround:
```sh
$ touch 'foo-and-bar'
$ ls
foo-and-bar
```

This is similar to #1432 discussion on quoting/escaping of special characters; in particular whitespace. But I think this is a distinct issue and I **very** much hope that _this_ is not intended behaviour.

**Comment by scopatz** [comment]
Hi @OJFord - thanks for reporting.  This is not the intended behaviour. and, or, and not should only be keywords when surrounded by whitespace.

**Comment by anddam** [comment]
I have the same issue using **-or** with **find** on 0.5.0 .

**Comment by OJFord** [comment]
Yes, I had a fiddle with it, but I didn't have any luck without breaking other behaviour. I'm happy to have another look in the new year.

**Comment by scopatz** [comment]
Potential fix in #2274

**Comment by scopatz** [comment]
Closing this since #2274 has been merged. Please fee free to reopen if needed
